pkg/controllers/provisioning/controller.go

```diff
@@ -129,6 +131,19 @@ func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 		NewControllerManagedBy(m).
 		Named("provisioning").
 		For(&v1alpha5.Provisioner{}).
+		WithEventFilter(predicate.Funcs{
```

Comment:
Instead of this, what do you think about doing an `object.DeepEquals(provisoner.Spec, active.Spec)` and only restarting the provisioner if something has changed? We'd need to check the instance types as well.

Reply:
I'd be fine with either checks - the generation value is probably a little cheaper?
Not sure I follow the instance types - do you mean the fact that we restart the provisioner every 5 minutes in any case to refresh our knowledge of ICEs?